app/controllers/jobseekers/job_applications/build_controller.rb

```diff
@@ -134,10 +134,17 @@ def trn_params
     }
   end
 
-  def update_jobseeker_profile
-    current_jobseeker.jobseeker_profile.update(
-      teacher_reference_number: form_params[:teacher_reference_number],
-      has_teacher_reference_number: form_params[:has_teacher_reference_number],
-    )
+  def update_or_create_jobseeker_profile
+    if current_jobseeker.jobseeker_profile.nil?
+      current_jobseeker.create_jobseeker_profile(
```

Comment:
I think this needs a ! at the end - is there enough data here to create a profile?

Reply:
I'm not sure. Both personal_details and job_preferences are has_one w/o optional - I *think* that rails will insist on them?